src/encoded/types/shared_calculated_properties.py

```diff
@@ -482,85 +482,85 @@ class CalculatedReplicationType:
         "description": "Calculated field that indicates the replication model",
         "type": "string"
     })
-    def replication_type(self, request, replicates=None, assay_term_name=None):
+    def replication_type(self, request, original_files=None, replicates=None, assay_term_name=None):
         # ENCD-4251 loop through replicates and select one replicate, which has
         # the smallest technical_replicate_number, per biological replicate.
         # That replicate should have a libraries property which, as calculated
         # in replicate.libraries (ENCD-4251), should have collected all
         # possible technical replicates belong to the biological replicate.
         # TODO: change this once we remove technical_replicate_number.
+        # replication_type for experiment without data files is not defined
+        if not original_files:
+            return None
+        # replication_type for experiment without replicates is not defined
+        if not replicates:
+            return None
+        excluded_statuses = (
+            'uploading',
+            'content error',
+            'upload failed',
+            'deleted',
+            'replaced')
+        filtered_file_objects = [
+            request.embed(f, '@@object')
+            for f in paths_filtered_by_status(
+                    request,
+                    original_files,
+                    exclude=excluded_statuses
+            )
+        ]
+        related_replicates_from_filtered_files = [
+            f.get('replicate')
+            for f in filtered_file_objects 
+            if f.get('replicate') and f.get('replicate') in replicates
+        ]
+        filtered_replicate_objects = [
+            request.embed(r, '@@object')
+            for r in paths_filtered_by_status(
+                    request,
+                    related_replicates_from_filtered_files,
+                    exclude=excluded_statuses
```

Comment:
I think there is a similar indentation issue going on here, this also should have trailing comma.
```suggestion
            for r in paths_filtered_by_status(
                request,
                related_replicates_from_filtered_files,
                exclude=excluded_statuses,
```

Reply:
fixed